src/commands/createFunctionApp/FunctionAppCreateStep.ts

```diff
@@ -97,11 +100,57 @@ export class FunctionAppCreateStep extends AzureWizardExecuteStep<IFunctionAppWi
         site.extendedLocation = { name: customLocation.id, type: 'customLocation' };
     }
 
-    private async getNewSiteConfig(context: IFunctionAppWizardContext, stack: FullFunctionAppStack): Promise<SiteConfig> {
-        const stackSettings: FunctionAppRuntimeSettings = nonNullProp(stack.minorVersion.stackSettings, context.newSiteOS === WebsiteOS.linux ? 'linuxRuntimeSettings' : 'windowsRuntimeSettings');
-        const newSiteConfig: SiteConfig = stackSettings.siteConfigPropertiesDictionary;
-        const storageConnectionString: string = (await getStorageConnectionString(context)).connectionString;
+    private async getNewFlexSite(context: IFunctionAppWizardContext): Promise<Site> {
+        const location = await LocationListStep.getLocation(context, webProvider);
+        const site: Site & { properties: FlexFunctionAppProperties } = {
+            name: context.newSiteName,
+            kind: getSiteKind(context),
+            location: nonNullProp(location, 'name'),
+            properties: {
+                name: context.newSiteName,
+                serverFarmId: context.plan?.id,
+                clientAffinityEnabled: false,
+                siteConfig: await this.getNewSiteConfig(context),
+                reserved: context.newSiteOS === WebsiteOS.linux  // The secret property - must be set to true to make it a Linux plan. Confirmed by the team who owns this API.
+            },
+        };
 
+        // TOOD: hardcode list of locations
+        site.location = 'North Central US (Stage)';
```

Comment:
Will this value eventually be taken from the context?  It looks like you may have already set it on the context in another spot, so wondering if you can just read it off the context at this point

Reply:
It should actually be taken from a `LocationListStep`, but like I mentioned above, the locations don't work.

I will add a PR to get the `LocationListStep` actually prompting with the proper location list soon though so I can remove this line and things should just work.